src/components/ProjectSearch/TextSearch.js

```diff
@@ -38,6 +39,19 @@ export default class TextSearch extends Component {
     shortcuts.off("Enter", this.onEnterPress);
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    if (this.props.query === "") {
+      return;
+    }
+
+    const resultCount = this.getResultCount();
+    if (resultCount === prevState.resultCount) {
+      return;
+    }
+
+    this.setState({ resultCount });
```

Comment:
do we need to store resultCount? why can't we use it in the render method... perhaps, `this.getResultCount` should return null if `this.props.query === ""` or we don't render the summary if `this.props.query === ""`